src/components/App.tsx

```diff
@@ -55,6 +66,44 @@ export default function App(){
 }
 
 function _App() {
+	// initial auth state
+	const [auth, setAuth] = useState<INavigatorProps>({
+		shouldSetup: false,
+		shouldAuth: ''
+	})
+	// app was longer than 5 mins in the background
+	const [bgAuth, setBgAuth] = useState(false)
+	// PIN mismatch state
+	const [attempts, setAttempts] = useState({
+		mismatch: false,
+		mismatchCount: 0,
+		locked: false,
+		lockedCount: 0,
+		lockedTime: 0,
+	})
+	const handlePinForeground = async () => {
+		// check if app is locked
+		const now = Math.ceil(Date.now() / 1000)
+		const lockData = await store.getObj<ILockData>('lock')
+		if (lockData) {
+			// set state acccording to lockData timestamp
+			const secsPassed = now - lockData.timestamp
+			const lockedTime = lockData.lockedTime - secsPassed
+			setAttempts({
+				...lockData,
+				mismatch: false,
+				lockedTime
+			})
+		}
+		// handle app was longer than 5 mins in the background
+		const bgTimestamp = await store.get('authBg')
```

Comment:
pls prefix all store keys 
like you do for settings 
await store.set('settings:lang', code)

Reply:
done